src/detect-engine-build.c

```diff
@@ -408,7 +408,9 @@ PacketCreateMask(Packet *p, SignatureMask *mask, AppProto alproto,
         (*mask) |= SIG_MASK_REQUIRE_NO_PAYLOAD;
     }
 
-    if (p->events.cnt > 0 || app_decoder_events != 0 || p->app_layer_events != NULL) {
+    if (p->events.cnt > 0 || app_decoder_events ||
+            (p->app_layer_events != NULL && p->app_layer_events->cnt) ||
```

Comment:
wonder if we should set a flag in `Packet::flags` on setting an event in either of these, to avoid the overhead of these checks (that also happen in logging I think)
`PKT_HAS_APP_EVENTS` or something like it

Reply:
@jlucovsky ^

Reply:
Yes -- good idea.